android/app/src/main/java/com/zulipmobile/sharing/ReceiveShareActivity.kt

```diff
@@ -0,0 +1,129 @@
+package com.zulipmobile.sharing
+
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
+import android.net.Uri
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.Log
+import android.webkit.WebView
+import androidx.annotation.Nullable
+import com.facebook.react.ReactActivity
+import com.facebook.react.ReactApplication
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.Arguments
+import com.zulipmobile.MainActivity
+import com.zulipmobile.notifications.*
+
+const val TAG = "ZulipReceiveShare"
+
+class ReceiveShareActivity : ReactActivity() {
+
+    /**
+     * Returns the name of the main component registered from JavaScript.
+     * This is used to schedule rendering of the component.
+     */
+    override fun getMainComponentName(): String? {
+        return "SharingRoot"
+    }
+
+    private fun sendEvent(reactContext: ReactContext,
+                          eventName: String,
+                          @Nullable params: WritableMap) {
+        Log.d(TAG, "Sending event with shared data")
+        emit(reactContext, eventName, params)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        WebView.setWebContentsDebuggingEnabled(true)
+        val application = application as ReactApplication
+        launchMainActivity(application as Context)
+        val host = application.reactNativeHost
+        val reactContext = host.tryGetReactInstanceManager()?.currentReactContext
+        val intent = intent
+        when {
+            intent?.action == Intent.ACTION_SEND -> {
```

Comment:
I felt so too! That is how I had originally implemented it. But Android Studio suggested to convert it to `when`, so I thought that's the proper way to do it in Kotlin.

Reply:
Was this with one of the "intentions" that you get behind a lightbulb icon in the left margin, and that you can see by hitting Alt+Enter?

I think Android Studio provides some of those that go in both directions :smile: . I think this may be one of them. Sometimes they're more in the vein of refactoring tools, like "ways you might refactor the code, depending on what you want and the context" -- rather than lint rules with fixers, like "ways the code is unidiomatic or bad and you can fix it".

Reply:
I got a green squiggle underline with this popup:
![image](https://user-images.githubusercontent.com/7714968/83006520-f7d41580-a02f-11ea-80b5-a7201cea2900.png)

Reply:
Ah, that looks like the other `when`, though :slightly_smiling_face: . In particular "Cascade if" describes the inner `when`, the one with several different options, but not the outer one which has just one option.

Reply:
(And yeah, that one isn't the "intentions" thing I was thinking of -- it's definitely saying something in the spirit of "this code is unidiomatic or bad and you should do it this other way".)